### PR TITLE
Add option to load specific languages instead of all

### PR DESCRIPTION
**Pull request by LaurentValette**
Proposed change:
* Add optional argument `languages` to the `load()` function.
* For projects that only work with specific languages, this reduces loading times and memory consumption.
* It is compatible with existing code, because the behaviour does not change when passing no arguments (loads all languages).

### CODE REVIEW

**Comment by FraBle** [comment]
Thanks for your PR, I will look into it!

File: duckling/language.py [hunk 1]
```diff
@@ -58,3 +58,13 @@ def convert_to_duckling_language_id(cls, lang):
         else:
             raise ValueError("Unsupported language '{}'. Supported languages: {}".format(
                 lang, ", ".join(cls.SUPPORTED_LANGUAGES)))
+
+    @classmethod
+    def convert_to_iso(cls, lang):
```
Comment (FraBle) [inline duckling/language.py]: Why aren't you reusing `convert_to_duckling_language_id`? Or replace everything with `convert_to_iso`? `convert_to_iso` and `convert_to_duckling_language_id` are basically the same function.

File: duckling/duckling.py [hunk 2]
```diff
@@ -78,10 +78,33 @@ def _create_classpath(self):
...
+                e.g. [Duckling.ENGLISH, Duckling.FRENCH] or supported ISO 639-1 Codes (e.g. ["en", "fr"])
+        """
         duckling_load = self.clojure.var("duckling.core", "load!")
-        duckling_load.invoke()
+
+        if languages is not None:
+            # Duckling's load function expects ISO 639-1 Language Codes (e.g. "en")
+            languages = [Language.convert_to_iso(lang) for lang in languages]
+
+            clojure_hashmap = self.clojure.var("clojure.core", "hash-map")
```
Comment (FraBle) [inline duckling/duckling.py]: `clojure_hashmap` , `clojure_list` can be. moved up to line 90, so it's after `duckling_load`

File: duckling/duckling.py [hunk 3]
```diff
@@ -78,10 +78,33 @@ def _create_classpath(self):
                     jars.append(os.path.join(top, file_name))
         return os.pathsep.join(jars)
 
-    def load(self):
-        """Loads the Duckling corpus"""
+    def load(self, languages=None):
```
Comment (FraBle) [inline duckling/duckling.py]: `languages` should be an empty list by default

File: duckling/duckling.py [hunk 4]
```diff
@@ -78,10 +78,33 @@ def _create_classpath(self):
...
+        Languages can be specified, defaults to all.
+
+        Args:
+            languages: Optional parameter to specify languages,
+                e.g. [Duckling.ENGLISH, Duckling.FRENCH] or supported ISO 639-1 Codes (e.g. ["en", "fr"])
+        """
         duckling_load = self.clojure.var("duckling.core", "load!")
-        duckling_load.invoke()
+
+        if languages is not None:
```
Comment (FraBle) [inline duckling/duckling.py]: You should turn around the if-else statement:
```python
if languages:
    ...
else:
    ...
```

File: README.md [hunk 5]
```diff
@@ -209,8 +209,15 @@ Duckling(jvm_started=False, parse_datetime=False, minimum_heap_size='128m', maxi
             is 2048m.
     """
 
-duckling.load():
-        """Loads the Duckling corpus"""
+duckling.load(self, languages=None):
```
Comment (FraBle) [inline README.md]: Please keep in mind if you change the API, that you update the README as well.
For the future, I plan to generate the docs via Sphinx, so this step won't be necessary anymore. 
Unfortunately, we still have to do it for now 🙁

File: duckling/language.py (hunk 1 above)
Comment (LaurentValette) [inline duckling/language.py]: The reasoning here was that the `load` function of duckling supports iso codes only. But, yes, that could definitely be cleaned. I'll fix that.

Comment (FraBle) [inline duckling/language.py]: Thanks!

**Comment by FraBle** [comment]
Merged and released as [version 1.4.0](https://pypi.python.org/pypi/duckling).

**Pull request merged**